test/nullable.jl

```diff
@@ -279,3 +279,14 @@ end
 
 # issue #11675
 @test repr(Nullable()) == "Nullable{Union{}}()"
+
+# Ensure that promotion works
+types = [
+    (DateTime, DateTime),
+    (Float64, Int64),
+]
+
+for (S, T) in types
+    @test Base.promote_type(Nullable{S}, Nullable{T}) == Nullable{Base.promote_type(S, T)}
```

Comment:
`promote_type` is tested above now, so you could as well remove the loop and simply copy the line for a few types. Would make sense to put it right below the `promote_type` block.

Also, it would be good to test the actual behaviour, i.e addition, subtraction, etc. of nullables.